### DISCUSSION TITLE
"Missing x-api-key" error when it is present and being accessed syntactically correct

### DISCUSSION THREAD

**Discussion by blipchin26** [etsy/open-api#1029] [Q&A] [answered]
The attached code is how I am trying to find customer data on Etsy, but every time I run the code I receive a 401 Client Error. I am not sure how this is occurring as I thought I was accessing the URL correctly with my headers and parameters. If anyone could help me figure out what I am doing wrong it would be greatly appreciated!

![github q](https://github.com/etsy/open-api/assets/113556181/f981545c-5d2e-4352-8dfe-90b3426c8e36)

**Comment by Gareth064** [comment, marked as answer]
Doesn't look like you have done any OAuth2 authentication to get an access token for the shop you're trying to access. 

This is documented on Etsy's Developer portal if you want to check there.
